pkg/controller/master/backup/backup_metadata.go

```diff
@@ -84,14 +85,21 @@ func (h *MetadataHandler) OnBackupTargetChange(key string, setting *harvesterv1.
 		return nil, nil
 	}
 
-	target, err := settings.DecodeBackupTarget(settings.BackupTargetSet.Get())
+	target, err := settings.DecodeBackupTarget(setting.Value)
```

Comment:
Will it return json unmarshal error when `setting.Value` is empty?

Reply:
setting.Value is checked nil before calling Decode,  Decode itself does not check if/not value is nil.

```
	if setting == nil || setting.DeletionTimestamp != nil ||
		setting.Name != settings.BackupTargetSettingName || setting.Value == "" {
		return nil, nil
	}

	-target, err := settings.DecodeBackupTarget(settings.BackupTargetSet.Get())
	//target, err := settings.DecodeBackupTarget(settings.BackupTargetSet.Get())
	+target, err := settings.DecodeBackupTarget(setting.Value)
```